containerd-shim-spin/src/engine.rs

```diff
@@ -397,6 +405,15 @@ impl SpinEngine {
     }
 }
 
+// For each container environment variable, duplicates it in the environment
+// with a given prefix
+fn prefix_env_vars(prefix: &str) {
+    std::env::vars().for_each(|(var, val)| {
+        let prefixed = format!("{}{}", prefix, var);
+        std::env::set_var(prefixed, val);
```

Comment:
suggestion: if there are already env vars with the prefixed names, we should throw a warning that this function will overwrite them.

Reply:
```suggestion
        let prefixed = format!("{}_{}", prefix, var);
        if env::var(&prefixed).is_ok() {
            log::warn!("Overwriting existing variable: {}", prefixed);
        }
        env::set_var(prefixed, val)?;
```

Reply:
Thats a good point. I think in this case we shouldn't overwrite. Maybe instead we set if not exists